music21/instrument.py

```diff
@@ -128,11 +129,33 @@ class Instrument(base.Music21Object):
     * instrumentAbbreviation
     * midiProgram
     * midiChannel
-    * lowestNote (a note object or a string)
-    * highestNote (a note object or a string)
+    * lowestNote (a note object or a string for _written_ pitch)
+    * highestNote (a note object or a string for _written_ pitch)
```

Comment:
Interesting, I don't see that strings are used anywhere for these -- do you?

Reply:
Indeed! 

Clearly the description should match the object, though I'm inclined to think that the object should change. Myke's concern over the creation of pitches as part of start-up time would be improved by initialising this variable to pitch name stings only and maybe having a separate method for converting them to pitch objects when needed. I wonder how many people rely on lowestPitch at all, how many of them actually want a pitch and not simply the string, and whether any of those people would have an issue with adding such a line to their code for that conversion.

Apart from the type change, we should also re-name 'lowestNote' as 'lowestPitch'. This is a more accurate description, but again, that would need checking given existing usage.

Again, this is perhaps one for a separate PR.

Reply:
All this discussion and we all missed until now -- it's a pitch object not a note object!  :-)  

These will need to become cached-auto-convert strings to pitch objects in the future.  Will switch that at some point (probably for v7 since it'll be a breaking change)